frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -3590,6 +3596,17 @@ const IR::Node *TypeInference::postorder(IR::MethodCallExpression *expression) {
                 param = *paramIt;
             }
 
+            if (param->type->is<IR::Type_Dontcare>())
+                typeError("%1%: Could not infer a type for parameter %2%", arg, param);
```

Comment:
```suggestion
                typeError("%1%: Could not infer a type for parameter %2% because the parameter's type is \"Type_Dontcare\"", arg, param);
```

Reply:
Type_Dontcare is really an internal class name of the compiler which does not mean much to the user.
But we can say "_".

Reply:
Maybe instead of Type_Dontcare calling it a don't-care type can work? `_` also looks confusing.